ompi/mca/coll/han/coll_han_gather.c

```diff
@@ -417,8 +415,18 @@ mca_coll_han_gather_intra_simple(const void *sbuf, int scount,
             reorder_buf = (char *)malloc(rsize);
             /* rgap is the size of unused space at the start of the datatype */
             reorder_buf_start = reorder_buf - rgap;
-        }
 
+	    if (MPI_IN_PLACE == sbuf) {
```

Comment:
I don't think we need this extra buffer. If we have `sbuf == MPI_IN_PLACE` it should be enough to set `sbuf = rbuf` and let the rest of the code unfolds. This way we will be sending the data from the correct buffer (aka the one provided by the user on the original call), but we will not need an additional allocation.

Reply:
sbuf doesn't seem to be used after this point, do you mean set reorder_buf_start = sbuf?